### PR TITLE
Move MAX, MIN and ABS to contiki-macros

### PR DESCRIPTION
**Pull request by cmorty**
Based on the discussion in #878 I started moving MIN, MAX and ABS macros to an extra file.

### CODE REVIEW

**Comment by nvt** [comment]
:+1:

**Comment by g-oikonomou** [comment]
Where are we on this one? I mean it would naturally need rebased, but it'd be great to get an update (?)

**Comment by bthebaudeau** [comment]
It would be great to merge this one now (after rebasing if needed), and to create a new PR later for #878.

**Comment by cmorty** [comment]
Getting this merged rather soon would be nice. Had to fix two new files.

**Comment by bthebaudeau** [comment]
Sorry, rebase needed.

**Comment by cmorty** [comment]
There you go. Rebase seems find. :)

File: platform/cooja/contiki-cooja-main.c [hunk 1]
```diff
@@ -39,7 +39,8 @@
 #include <stdio.h>
 #include <string.h>
 
-#include "contiki.h"
+#include <contiki.h>
```
Comment (bthebaudeau) [inline platform/cooja/contiki-cooja-main.c]: According to what @adamdunkels said, double quotes should be used here, not angle brackets.

**Comment by bthebaudeau** [comment]
@adamdunkels: In the end, are you happy with the naming? You did not say if you finally agreed with @cmorty arguments. I vote for simplicity here: a prefix would bring no benefits.

File: platform/cooja/contiki-cooja-main.c (hunk 1 above)
Comment (cmorty) [inline platform/cooja/contiki-cooja-main.c]: That one slipped through. I'll look into it

**Comment by bthebaudeau** [comment]
:+1: from me.

**Comment by simonduq** [comment]
and another :+1:

**Comment by adamdunkels** [comment]
:+1:

**Pull request merged**